filament/backend/src/metal/MetalHandles.h

```diff
@@ -119,8 +119,14 @@ class MetalBufferObject : public HwBufferObject {
     void updateBuffer(void* data, size_t size, uint32_t byteOffset);
     MetalBuffer* getBuffer() { return &buffer; }
 
+    // Flags that this BufferObject is used as a uniform buffer.
+    // If this isn't set, we can skip a check when destroying this BufferObject.
+    void flagUniformUsage() { isUniform = true; }
+    bool isUsedAsUniform() const { return isUniform; }
+
 private:
     MetalBuffer buffer;
+    bool isUniform;
```

Comment:
instead of a bool, you could use a `bitset32` and set the `index` bit when you bind (and unset it when you unbind) -- this way you could avoid to loop on the whole list, since each bit set would tel you which index has this ubo bound to.
We even have a `forEachBitSet` method you could use instead of the loop.

Reply:
Good suggestion, I like that